crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -741,6 +741,41 @@ impl AccountData {
     }
 }
 
+/// Single entry for a room-related read receipt configuration in `ReceiptsConfig`.
+#[derive(Clone, Debug, PartialEq)]
+pub enum RoomReceiptConfig {
+    /// Get read receipts for all the subscribed rooms.
+    AllSubscribed,
+    /// Get read receipts for this particular room.
+    Room(OwnedRoomId),
+}
+
+impl Serialize for RoomReceiptConfig {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            RoomReceiptConfig::AllSubscribed => serializer.serialize_str("*"),
+            RoomReceiptConfig::Room(r) => r.serialize(serializer),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for RoomReceiptConfig {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        match String::deserialize(deserializer)?.as_str() {
```

Comment:
You can use `deserialize_cow_str` from `ruma_common::serde` here for a bit more efficiency.